packages/jest-cli/src/runTest.js

```diff
@@ -13,74 +13,91 @@ import type {Path, Config} from 'types/Config';
 import type {TestResult} from 'types/TestResult';
 import type {Resolver} from 'types/Resolve';
 
+const fs = require('fs');
+const {getTestEnvironment} = require('jest-config');
 const BufferedConsole = require('./lib/BufferedConsole');
+const promisify = require('./lib/promisify');
 const {
   Console,
   NullConsole,
   setGlobal,
 } = require('jest-util');
 const getConsoleOutput = require('./reporters/getConsoleOutput');
 
+const testEnvironmentPragmaRegex = /@testEnvironment\s+(.*)/;
+
 function runTest(path: Path, config: Config, resolver: Resolver) {
-  /* $FlowFixMe */
-  const TestEnvironment = require(config.testEnvironment);
-  /* $FlowFixMe */
-  const TestRunner = require(config.testRunner);
-  /* $FlowFixMe */
-  const ModuleLoader = require(config.moduleLoader || 'jest-runtime');
+  return promisify(fs.readFile)(path, 'utf8').then(data => {
```

Comment:
Can you switch this to `fs.readFileSync`? 

I'm a bit worried about the fact that this will end up reading every test file twice and we could potentially change things around by passing a fake FS into `ModuleLoader` that has a cache, like this:

new ModuleLoader({
  …,
  cacheFS: {
    [path]: source,
  },
});

and then later pass that source to `transform` inside of `jest-runtime` so that we don't read the same file twice. What do you think?